src/writer.rs

```diff
@@ -182,7 +176,7 @@ where
     };
 
     let mut header = NiftiHeader {
-        dim,
+        dim: build_dim_array(data.shape()),
```

Comment:
At this level however, I question whether it should panic or error. It might be uncertain how much control the consumer has of the provided array. What do you think? If it should error instead, we can just add a some kind of test like the one below and make it pass.

```rust
let path = get_temporary_path(path);
let data = ArrayD::zeros(&[1; 8]);
assert!(write_nifti(path, data, None).is_err());
```

Reply:
Good questions. I don't really want this function to be public, but the outside tests also use it, so this might be an indication that this function can be useful to others.

I think the best is to be consistant. We already have `validate_dim` and `validate_dimensionality`, which return an error. I'll move the new function there, rename it/them and make it return an error.

Reply:
Oh by the way, we already have [`Shape::from_slice`](https://github.com/Enet4/nifti-rs/blob/master/src/volume/shape.rs#L140), which takes a slice of `u16`. With a few num traits, we should be able to make it generic for any slice of unsigned numbers.